graphx/src/main/scala/org/apache/spark/graphx/util/PeriodicGraphCheckpointer.scala

```diff
@@ -87,10 +88,7 @@ private[mllib] class PeriodicGraphCheckpointer[VD, ED](
 
   override protected def persist(data: Graph[VD, ED]): Unit = {
     if (data.vertices.getStorageLevel == StorageLevel.NONE) {
-      data.vertices.persist()
-    }
-    if (data.edges.getStorageLevel == StorageLevel.NONE) {
-      data.edges.persist()
+      data.persist()
```

Comment:
I enhanced the persistence tests in `PeriodicGraphCheckpointSuite` to check that the storage level requested in the graph construction is the storage level seen after persistence. Both this version and the original version of this method failed that unit test.

The graph's vertex and edge rdds are somewhat peculiar in that `.cache()` and `.persist()` do not do the same thing, unlike other RDDs. And while `.cache()` honors the default storage level specified at graph construction time, `.persist()` always caches with the `MEMORY_ONLY` storage level.

At any rate, getting the `PeriodicGraphCheckpointer` to honor the default storage level specified at graph construction time requires changing these method calls from `persist()` to `cache()`.